frontend/src/utils.ts

```diff
@@ -116,3 +116,85 @@ export class Emitters {
   // Coffee Chats
   static coffeeChatsUpdated: EventEmitter<void> = new EventEmitter();
 }
+
+/**
+ * Parses links from english text and fixes a provided link to ensure it adheres to a proper format for LinkedIn or GitHub profiles.
+ * Also handles raw usernames (e.g., "jujulcrane") by constructing a proper URL.
+ *
+ * @param {string} link - The input link, which could be a full URL, partial URL, or raw username.
+ * @param {boolean} git - Whether the link is for GitHub. If true, it processes the link as a GitHub URL.
+ * @param {boolean} linkedIn - Whether the link is for LinkedIn. If true, it processes the link as a LinkedIn URL.
+ * @returns {string | undefined} - The fixed and properly formatted link or undefined if the input is invalid.
+ *
+ * @example
+ * fixLink("https://linkedin.com/in/jujulcrane", false, true);
+ * // Returns: "https://www.linkedin.com/in/jujulcrane"
+ *
+ * @example
+ * fixLink("github.com/jujulcrane", true, false);
+ * // Returns: "https://github.com/jujulcrane"
+ *
+ * @example
+ * fixLink("Visit my profiles: https://linkedin.com/in/jujulcrane and github.com/jujulcrane", false, true);
+ * // Returns: undefined (multiple links in the input)
+ *
+ * @example
+ * fixLink("jujulcrane", false, true);
+ * // Returns: "https://www.linkedin.com/in/jujulcrane/"
+ */
+export const fixLink = (link: string, git: boolean, linkedIn: boolean): string | undefined => {
+  if (!link) {
+    return undefined;
+  }
+
+  const urlRegex =
+    /(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_+.~#?&//=]*)/g;
+  const matches = link.match(urlRegex);
+
+  if (matches && matches.length > 1) {
+    return undefined;
+  }
+
+  if (!matches) {
+    if (linkedIn) {
+      return `https://www.linkedin.com/in/${link.trim().toLowerCase()}/`;
+    }
+    if (git) {
+      return `https://github.com/${link.trim().toLowerCase()}`;
+    }
+    return undefined;
+  }
+
+  const extractedLink = matches[0]
+    .trim()
+    .replace(/[.,)]+$/, '')
+    .toLowerCase();
+
+  if (linkedIn) {
```

Comment:
This whole bottom section feels like it doesn't need to split into the LinkedIn and github links tbh. Since the only case where it matters which link type it is is if they just give us a username, which is handled above. Otherwise, I think the logic here can be as simple as checking if it's prefixed by `https://`, and if not, adding that. Do let me know if I'm missing a case though!